File: CheckingAccount.java
public class CheckingAccount extends BankAccount {

    //a subclass of BankAccount

    public static int transactionCount =0;        //make it a protected object?   also, why errors in package?
    private static final double TRANSACTION_FEE = 3.00;

    public CheckingAccount(String n, double b) {
        super(n, b);
    }

    public void deductFees() {
        double fee = transactionCount * TRANSACTION_FEE;
        this.withdraw(fee);
        transactionCount = 0;
    }
}

File: CheckingAccount.cpp
//
//  CheckingAccount.cpp
//  BankAccount
//
//  Created by Rebekah on 11/20/18.
//  Copyright © 2018 Rebekah. All rights reserved.
//

#include "CheckingAccount.hpp"
#include "BankAccount.hpp"

/*
 a constructor for CheckingAccount
 */
CheckingAccount::CheckingAccount(string n, double b){
    this-> name = n;
    this -> balance = b;
}

/*
 calls the withdraw method from BankAccount
 */
void CheckingAccount::withdraw(double a){
    BankAccount::withdraw(a);
    transactionCount++;
}

/*
 calls the deposit method from BankAccount
 */
void CheckingAccount::deposit(double a){
    BankAccount::deposit(a);
    transactionCount++;
}

/*
 deducts the transaction fees from a CheckingAccount
 */
void CheckingAccount::deductFees(){
    withdraw(transactionCount*TRANSACTION_FEE);
    transactionCount = 0;
}
